### PR TITLE
For issue #18132 remove icons from settings page

### PR DESCRIPTION
**Pull request by Amejia481**
This PR removes the icons from the setting page and put it behind a feature flag only active in debug and nightly. Below a view of the end result.

https://user-images.githubusercontent.com/773158/110135493-969a5900-7d9c-11eb-830d-ebba5c2ad3bd.mp4

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -132,7 +133,12 @@ class SettingsFragment : PreferenceFragmentCompat() {
     }
 
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
-        setPreferencesFromResource(R.xml.preferences, rootKey)
+        val preferencesId = if (FeatureFlags.newIconSet) {
+            R.xml.preferences_without_icons
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:137]: Perfect! 👌

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt [hunk 2]
```diff
@@ -368,8 +374,10 @@ class SettingsFragment : PreferenceFragmentCompat() {
         val preferenceOpenLinksInExternalApp =
             findPreference<Preference>(getPreferenceKey(R.string.pref_key_open_links_in_external_app))
 
-        preferencePrivateBrowsing.icon.mutate().apply {
-            setTint(requireContext().getColorFromAttr(R.attr.primaryText))
+        if (!FeatureFlags.newIconSet) {
+            preferencePrivateBrowsing.icon.mutate().apply {
+                setTint(requireContext().getColorFromAttr(R.attr.primaryText))
+            }
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:380]: Why do we need to use `.mutate().apply` instead of just `.apply`?

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:380]: I'm not really sure, I didn't added initially I just moved the line XD, I will investigated :)

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:380]: Thanks @cadeyrn  for the clarification :)

Comment (NotWoods) [reply app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:380]: Yup, otherwise any changes will affect other instances of the drawable

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:380]: Thanks 👋🏽

**Review by eliserichards — APPROVED** [review]
💯

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt (hunk 2 above)
Comment (NotWoods) [reply app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:380]: Miss y'all too!

**Comment by cadeyrn** [comment]
If you remove all the icons from the settings page should the reserved icon space not be removed as well? It looks really odd with no icons but such a large gap. For reference: Neither in Chrome nor in the Samsung browser there is such a large gap.

**Review by brampitoyo — APPROVED** [review]
For an initial removal, this looks good!

Our next (and final) step for Settings is to reduce the left-side margin/padding to be equal to the right-side, so that there’s no ‘missing’ empty space between the eliminated icon and the text.

I’m aware that it will cause some layout inconsistencies, especially when a menu item without an icon appears alongside a menu item with radio button on the left. That’s perfectly OK.

We can do this change if it’s easy, or file a follow-up issue if you’d like me to review all Settings page.

**Comment by Amejia481** [comment]
@brampitoyo thanks for taking a look I will try to do it in this PR but if it requires a lot of time I will address it on follow-up

**Comment by Amejia481** [comment]
@brampitoyo  I updated the PR removing the left-side margin/padding. I think we will need to rethink all the different states of the sync item like not logged in, logged in, error etc.
  
https://user-images.githubusercontent.com/773158/110876418-74756f00-82a5-11eb-8dbb-49a2b0a1f7c8.mp4

**Review by brampitoyo — APPROVED** [review]
@Amejia481 The removal of icons from Settings root page looks good to me, as it now aligns with desktop and iOS.

Have margins on other Settings sub-pages been tweaked, as well? We can do that in another issue.

**Comment by Amejia481** [comment]
> @Amejia481 The removal of icons from Settings root page looks good to me, as it now aligns with desktop and iOS.
> 
> Have margins on other Settings sub-pages been tweaked, as well? We can do that in another issue.

Thanks for the reviews.
No it hasn't been done, I will address it on another issue :)

**Comment by Imerion** [comment]
I must ask here as well, how come the icons in the options menu are to be removed? They help with finding the right option quickly, look good, makes it easier to keep track of movement when scrolling and gives the menu a professional and polished look. As far as I can see they are also still in the desktop version when going into Settings, and look good there as well. (Though I really with they were colorful and not flat.)

**Comment by Amejia481** [comment]
@brampitoyo could help to clarify your doubts.

**Comment by MPeti1** [comment]
I also don't think it's a good idea to remove the icons. I mean, remove it just because they will soon get removed from the desktop version of Firefox too? That redesign is a huge mistake on it's own, please don't follow them here :(

**Pull request merged**